code/utils.py

```diff
@@ -0,0 +1,377 @@
+        "sound_offset",
+        "Hardware sound offset",
+    ),  # EVENT
+    "Behavior.HarpBehavior.PulseSupplyPort0": (
+        ["PulseSupplyPort0"],
+        "water_onset",
+        "Hardware water onset",
+    ),  # EVENT
+    "Behavior.HarpOlfactometer.OdorValveState": (
+        ["Valve0", "Valve1", "Valve2"],
+        "odor_line_load",
+        (
+            "Loading of odor to line. "
+            "Doesn’t mean odor is presented to the mouse "
+            "but needs to happen before EndValve trigger"
+            "and defines what odor is "
+            "being presented."
+        ),
+    ),  # EVENT
+    "Behavior.HarpOlfactometer.EndValveState": (
+        ["EndValve0"],
+        "Odor",
+        "Odor onset (True) and Odor offset (False)",
+    ),  # EVENT
+    "Behavior.HarpLickometer.LickState": (
+        ["Channel0"],
+        "Lick_state",
+        "Lick onset (True) and Lick offset (False)",
+    ),  # EVENT
+    "Behavior.HarpBehavior.DigitalInputState": (
+        ["DIPort0"],
+        "Photodiode",
+        "Screen synchronization photodiode",
+    ),  # CONTINUOUS
+    "Behavior.HarpOlfactometer.Channel0ActualFlow": (
+        ["Channel0ActualFlow"],
+        "Channel0ActualFlow",
+        "Measure flow in channel 0",
+    ),  # CONTINUOUS
+    "Behavior.HarpOlfactometer.Channel1ActualFlow": (
+        ["Channel1ActualFlow"],
+        "Channel1ActualFlow",
+        "Measure flow in channel 1",
+    ),  # CONTINUOUS
+    "Behavior.HarpOlfactometer.Channel2ActualFlow": (
+        ["Channel2ActualFlow"],
+        "Channel2ActualFlow",
+        "Measure flow in channel 2",
+    ),  # CONTINUOUS
+    "Behavior.HarpOlfactometer.Channel3ActualFlow": (
+        ["Channel3ActualFlow"],
+        "Channel3ActualFlow",
+        "Measure flow in channel 3",
+    ),  # CONTINUOUS
+    "Behavior.HarpOlfactometer.Channel4ActualFlow": (
+        ["Channel4ActualFlow"],
+        "Channel4ActualFlow",
+        "Measure flow in channel 4",
+    ),  # CONTINUOUS
+    "Behavior.HarpSniffDetector.RawVoltage": (
+        ["RawVoltage"],
+        "Breathing",
+        "Breathing signal",
+    ),  # CONTINUOUS
+    "Behavior.HarpStepperDriver.AccumulatedSteps": (
+        ["Motor0, Motor1, Motor2, Motor3"],
+        "MotorPositions",
+        "The position of x, y1, y2, and z  of the lickspout and oder tube",
+    ),  # CONTINUOUS
+    "Behavior.HarpTreadmill.SensorData": (
+        [
+            "Encoder",
+            "Torque",
+            "TorqueLoadCurrent",
+        ],
+        "Treadmill",
+        "Continuous signal from treadmill",
+    ),  # CONTINUOUS
+    "Behavior.HarpEnvironmentSensor.SensorData": (
+        [
+            "Pressure",
+            "Temperature",
+            "Humidity",
+        ],
+        "Environment",
+        "Continuous signal from environment sensor",
+    ),  # CONTINUOUS
+}
+
+
+# ported from Tiffany's processing code
+def get_breathing_from_sniff_detector(nwb: pynwb.NWBFile) -> np.ndarray:
```

Comment:
is this generalizable to all harp or just VR foraging? Can this function be altered to include a field `Behavior.HarpSniffDetector.RawVoltage`? Asking b/c it seems to return a numpy array